### PR TITLE
VZ-8079: Update Keycloak to v20.0.1

### PR DESCRIPTION
**Pull request by bhatpmk**
This change upgrades Keycloak to 20.0.1 in Verrazzano. Additional information about some of the changes done in this PR, other than the Helm charts -

**1. platform-operator/controllers/verrazzano/component/keycloak/keycloak.go:** Other than updating the code to reflect the new location of kcadm.sh, a new function is added to delete the StatefulSet and the headless service created by the chart for Keycloak. This is required as the StatefulSet in the version 20.0.1 contains changes other than allowed configurations 'replicas', 'template', and 'updateStrategy'. With this change, the function to set the replica count for the StatefulSet is not needed and removed
**2. platform-operator/helm_config/overrides/keycloak-values.yaml:**
- command: overrides default entrypoint of the Keycloak container. The backward compatibility flags "spi-login-protocol-openid-connect-legacy-logout-redirect-uri" and spi-login-protocol-openid-connect-suppress-logout-confirmation-screen" allow Verrazzano to use redirect_uri on logout from the Verrazzano console. These flags will be removed in a follow-up change
- fullnameOverride and nameOverride to get the name of the pod and StatefulSet to have "keycloak" v/s "keycloakx" as defined in the chart
- JAVA_OPTS -Djgroups.dns.query and port 7800, for the High Availability, with JGroups discovery via DNS_PING

### CODE REVIEW

File: platform-operator/controllers/verrazzano/component/keycloak/keycloak.go [hunk 1]
```diff
@@ -1535,67 +1535,47 @@ func GetOverrides(object runtime.Object) interface{} {
...
-	if !found {
-		return nil
-	}
-
 	// Get the StatefulSet for Keycloak
-	client := ctx.Client()
+	ctxClient := ctx.Client()
 	statefulSet := appv1.StatefulSet{}
-	err = client.Get(context.TODO(), types.NamespacedName{Namespace: ComponentNamespace, Name: ComponentName}, &statefulSet)
+	err := ctxClient.Get(context.TODO(), types.NamespacedName{Namespace: ComponentNamespace, Name: ComponentName}, &statefulSet)
```
Comment (mgianatagh) [inline platform-operator/controllers/verrazzano/component/keycloak/keycloak.go:1551]: Is it an error if the statefulset to be deleted does not exist?  Should this log a warning and return success?

File: platform-operator/controllers/verrazzano/component/keycloak/keycloak.go [hunk 2]
```diff
@@ -1535,67 +1535,47 @@ func GetOverrides(object runtime.Object) interface{} {
...
+// deleteHeadlessService deletes the keycloak-headless service after deleting the StatefulSet, before the upgrade
+func deleteHeadlessService(ctx spi.ComponentContext) error {
+	keycloakComp := ctx.EffectiveCR().Spec.Components.Keycloak
+	if keycloakComp == nil {
+		return nil
+	}
+	// Get and delete the headless service associated with the StatefulSet
+	service := &corev1.Service{}
+	ctxClient := ctx.Client()
+	if err := ctxClient.Get(context.TODO(), types.NamespacedName{Namespace: ComponentNamespace, Name: headlessService}, service); err != nil {
```
Comment (mgianatagh) [inline platform-operator/controllers/verrazzano/component/keycloak/keycloak.go:1573]: Is it an error if the service to be deleted does not exist?  Should this log a warning and return success?

Comment (bhatpmk) [reply platform-operator/controllers/verrazzano/component/keycloak/keycloak.go:1577]: Done

**Review by pfmackin — DISMISSED** [review]

**Review by mgianatagh — APPROVED** [review]

**Review by michael-cico — APPROVED** [review]

**Review by lake-of-dreams — APPROVED** [review]

**Comment by bhatpmk** [comment]
VZ-8079

**Pull request merged**